tests/state_test_util.go

```diff
@@ -361,12 +361,12 @@ func simulateEthGasPrice(config *params.ChainConfig, json *stJSON) error {
 
 // simulate intrinsic gas amount for Ethereum
 func simulateEthIntrinsicGas(intrinsicGas uint64, data []byte, r params.Rules) (uint64, error) {
-	if r.IsIstanbul {
+	if r.IsIstanbul && !r.IsPrague {
```

Comment:
It seems a contradiction because L369 sets `Prague`. I didn't intend to change this initially. Can you explain how it was derived?

Reply:
If Kaia is operating as the equivalent of Prague, this simulating can be skip because `kaiaPayloadGas == ethPayloadGas`. 

https://github.com/ethereum/go-ethereum/blob/v1.14.11/core/state_transition.go#L91-L93
`Prague` is set in L369 in order to calculate the payload gas of Ethereum after Istanbul.

Reply:
I think `intrinsicGas - kaiaPayloadGas` always becomes zero. Before Prague, so, only the ether logic can be considered maybe.

Reply:
I see. How about this? I think this is more intrinsic.

```golang
func simulateEthIntrinsicGas(data []byte, contractCreation bool, r params.Rules) (uint64, error) {
	if r.IsIstanbul {
		r.IsPrague = true
	}
	return types.IntrinsicGas(data, nil, contractCreation, r)
}
```

Reply:
Looks nice, but can you consider changing the function name such as `useEthIntrinsicGas` because `simulate` seems bloated?